server/src/Assets/assets.service.ts

```diff
@@ -0,0 +1,55 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { Asset } from './entities/asset.entity';
+import { Repository } from 'typeorm';
+import { CreateAssetDto } from './dto/create-asset.dto';
+import { UpdateAssetDto } from './dto/update-asset.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+
+@Injectable()
+export class AssetsService {
+  constructor(
+    @InjectRepository(Asset) private assetsRepository: Repository<Asset>,
+  ) {}
+
+  async create(CreateAssetDto: CreateAssetDto) {
+    return this.assetsRepository.save(CreateAssetDto);
+  }
+
+  findAll() {
+    return this.assetsRepository.find();
+  }
+
+  async findOne(id: number) {
+    const asset = await this.assetsRepository.findOne({ id });
+    if (!asset) {
+      throw new HttpException(
+        { status: HttpStatus.NOT_FOUND, error: 'Asset id not found' },
+        HttpStatus.NOT_FOUND,
+      );
+    }
```

Comment:
i think this should be handled in the controller if possible
so return value here might really be something like `: Promise<Asset | null>`

Reply:
controllers generally handle errors so that they're all in one place (controllers could be hitting multiple services)
also, we'll want to add these errors to the request object so that it can automagically be put into our hosting service's (i.e. AWS) logs